### PR TITLE
Mtan

### PR DESCRIPTION
**Pull request by CharlesAuthier**

### CODE REVIEW

File: article/_posts/2019-07-25-MTAN.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+**Code available [here](https://github.com/lorenmt/mtan).**
+
+
+This paper proposes a *Multi-Task Attention Network* (MTAN), an architecture that can be trained end-to-end and can be built upon any feed-foward neural network, simple to implement, and parameter efficient.
+
+![](/article/images/MTAN/proposal.png)
+
+# Introduction 
+
+The authors wanted to perform multiple task simultaneously to avoid training a independat network for each task, not only this approch is more efficient  in terms of memory and inference speed, but also seem that related task shre informative visual features.
```
Comment (AntoineTheb) [inline article/_posts/2019-07-25-MTAN.md]: independent, share

File: article/_posts/2019-07-25-MTAN.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Compare to standar single-task learning, training multiple task whilst succesfully learning a shared representation poses two challenges: How to share in a network architecture, and How to balence the tasks between losses functions.
```
Comment (AntoineTheb) [inline article/_posts/2019-07-25-MTAN.md]: standard, balance

File: article/_posts/2019-07-25-MTAN.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    authors: "Shikun Liu, Edward Johns, Andrew J. Davison"
+    title:   "End-to-End Multi-Task Learning with Attention"
+    venue:   "Accepted at Computer Vision and Pattern Recognition (CVPR), 2019"
+pdf:  https://arxiv.org/pdf/1803.10704.pdf
+---
+
+**Code available [here](https://github.com/lorenmt/mtan).**
+
+
+This paper proposes a *Multi-Task Attention Network* (MTAN), an architecture that can be trained end-to-end and can be built upon any feed-foward neural network, simple to implement, and parameter efficient.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: suggestion: '... network, is simple to implement, as well as parameter efficient.'

File: article/_posts/2019-07-25-MTAN.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+![](/article/images/MTAN/proposal.png)
+
+# Introduction 
+
+The authors wanted to perform multiple task simultaneously to avoid training a independent network for each task, not only this approch is more efficient in terms of memory and inference speed, but also seem that related task share informative visual features.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: wanted -> want

Comment (djoerch) [reply article/_posts/2019-07-25-MTAN.md]: ... not only is this approach ..., but it also seems that related tasks share ...

File: article/_posts/2019-07-25-MTAN.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Compare to standard single-task learning, training multiple task whilst succesfully learning a shared representation poses two challenges: How to share in a network architecture, and How to balance the tasks between losses functions.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: Compared to ...

Comment (djoerch) [reply article/_posts/2019-07-25-MTAN.md]: ... loss functions.

File: article/_posts/2019-07-25-MTAN.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+![](/article/images/MTAN/architecture.png)
+
+MTAN consists of a single shared network, and *K* task-specific attention networks.
+The shared network can be designed based on the particular task, whilst each task-specific network consists of a set of attention modules, which link with the shared network.
+Therefore, the features in the shared network, and the soft attention masks, can be learned jointly to maximise the generalisation of the shared features across multiple tasks, whilst simultaneously maximising the task-specific performance due to the attention masks.
+
+## Task Specific Attention Module 
+
+The attention module is designed to allow the taskspecific network to learn task-related features, by applying a soft attention mask to the features in the shared network, with one attention mask per task per feature channel.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: task-specific

File: article/_posts/2019-07-25-MTAN.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+The task-specific features in this layer, are then computed by element-wise multiplication of the attention masks with the shared features.
+
+## Losses
+
+This architecture combine a total of three losses via a weighted linear combination.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: combines

File: article/_posts/2019-07-25-MTAN.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+## Task Specific Attention Module 
+
+The attention module is designed to allow the taskspecific network to learn task-related features, by applying a soft attention mask to the features in the shared network, with one attention mask per task per feature channel.
+The task-specific features in this layer, are then computed by element-wise multiplication of the attention masks with the shared features.
+
+## Losses
+
+This architecture combine a total of three losses via a weighted linear combination. 
+The first loss is for the semantic segmentation, and it's a simply cross-entropy loss for each classes.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: simply -> simple

Comment (djoerch) [reply article/_posts/2019-07-25-MTAN.md]: 'for each class' / 'over all classes' ?

File: article/_posts/2019-07-25-MTAN.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+The secound loss is for the depth estimation, where they use a $$L_1$$ norm comparing the predicted and the ground-truth depth.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: ... use an $$L_1$$ ...

File: article/_posts/2019-07-25-MTAN.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+The last one is for the surfarce normals, only use for the NYUv2 dataset, where an element-wise dot product at each normalised pixel with the ground-truth map.
```
Comment (djoerch) [inline article/_posts/2019-07-25-MTAN.md]: typo -> surface

The subclause after 'where' is lacking a verb.

File: article/_posts/2019-07-25-MTAN.md (hunk 4 above)
Comment (jodoin) [inline article/_posts/2019-07-25-MTAN.md]: a independent -> an independent

Comment (jodoin) [reply article/_posts/2019-07-25-MTAN.md]: multiple tasks

File: article/_posts/2019-07-25-MTAN.md (hunk 5 above)
Comment (jodoin) [reply article/_posts/2019-07-25-MTAN.md]: and how to balance

File: article/_posts/2019-07-25-MTAN.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+The proposed configuration allows image-to-image dense pixel-level prediction, like semantic segmentation, depth estimation, and surface normal prediction.
+
+# Multi-Task Attention Network
+
+## Architecture Design
+
+![](/article/images/MTAN/architecture.png)
+
+MTAN consists of a single shared network, and *K* task-specific attention networks.
+The shared network can be designed based on the particular task, whilst each task-specific network consists of a set of attention modules, which link with the shared network.
```
Comment (jodoin) [inline article/_posts/2019-07-25-MTAN.md]: attention modules linked with the...

File: article/_posts/2019-07-25-MTAN.md (hunk 9 above)
Comment (jodoin) [inline article/_posts/2019-07-25-MTAN.md]: secound

Comment (jodoin) [inline article/_posts/2019-07-25-MTAN.md]: secound -> second

**Pull request merged**